src/System.Net.HttpListener/src/System/Net/Managed/HttpResponseStream.Managed.cs

```diff
@@ -150,6 +174,22 @@ internal void InternalWrite(byte[] buffer, int offset, int count)
             }
         }
 
+        internal async Task InternalWriteAsync(byte[] buffer, int offset, int count)
+        {
+            if (_ignore_errors)
+            {
+                try
+                {
+                    await _stream.WriteAsync(buffer, offset, count);
+                }
+                catch { }
+            }
+            else
+            {
+                await _stream.WriteAsync(buffer, offset, count);
```

Comment:
async methods have several allocations involved in their execution, but these allocations are only needed if the async method actually yields, meaning if any of its awaits await something that's not yet completed.  That means async methods aren't "free", in that each additional one in a chain of methods that may yield could result in more allocations.  If A() awaits B() awaits C() awaits D(), and D() doesn't complete synchronously, then A, B, and C will all incur those additional allocations.  By doing a transformation like the one I suggested, if !_ignore_errors, then you can avoid an extra layer of such awaiting and thus an extra layer of allocations.